### PR TITLE
Update background tasks guide

### PR DESCRIPTION
**Pull request by laktek**

### CODE REVIEW

File: apps/docs/content/guides/functions/background-tasks.mdx [hunk 1]
```diff
@@ -3,39 +3,59 @@ id: 'function-background-tasks'
 title: 'Background Tasks'
 description: 'How to run background tasks in an Edge Function outside of the request handler'
 subtitle: 'How to run background tasks in an Edge Function outside of the request handler'
-tocVideo: 'rSKBTJxG9VA'
```
Comment (laktek) [inline apps/docs/content/guides/functions/background-tasks.mdx:6]: @thorwebdev I removed the video from the guide as the syntax has changed (no longer need to do custom events). Would you be able to do an updated version of the video?

File: apps/docs/content/guides/functions/background-tasks.mdx [hunk 2]
```diff
@@ -3,39 +3,59 @@ id: 'function-background-tasks'
...
 
 Edge Function instances can process background tasks outside of the request handler. Background tasks are useful for asynchronous operations like uploading a file to Storage, updating a database, or sending events to a logging service. You can respond to the request immediately and leave the task running in the background.
 
-### How long a background task can run
+### How it works
 
-A background task can run until the Edge Function instance hits its wall-clock limit or reaches CPU/Memory limit. Check [limits](/docs/guides/functions/limits) for current caps.
+You can use `EdgeRuntime.waitUntil(promise)` to explicitly mark background tasks. Function instance would continue to run until the promise provided to `waitUntil` completes.
+
+The maximum duration it will run will be based on wall-clock, CPU or memory limit. The Function will shutdown when it reaches one of the [limits](/docs/guides/functions/limits).
```
Comment (charislam) [inline apps/docs/content/guides/functions/background-tasks.mdx:14]:
```suggestion
The maximum duration is capped based on the wall-clock, CPU, and memory limits. The Function will shutdown when it reaches one of these [limits](/docs/guides/functions/limits).
```

File: apps/docs/content/guides/functions/background-tasks.mdx [hunk 3]
```diff
@@ -3,39 +3,59 @@ id: 'function-background-tasks'
...
+
+You can listen to `beforeunload` event handler to be notified when function invocation is about to be shutdown.
```
Comment (charislam) [inline apps/docs/content/guides/functions/background-tasks.mdx:16]:
```suggestion
You can listen to the `beforeunload` event handler to be notified when Function invocation is about to be shut down.
```

File: apps/docs/content/guides/functions/background-tasks.mdx [hunk 4]
```diff
@@ -3,39 +3,59 @@ id: 'function-background-tasks'
...
 
 ### Example
 
-Here's an example of defining a background task using a custom event.
+Here's an example of using `EdgeRuntime.waitUntil` to run a background task and using `beforeunload` event to be notified when the instance is about to be shutdown.
```
Comment (charislam) [inline apps/docs/content/guides/functions/background-tasks.mdx:20]:
```suggestion
Here's an example of using `EdgeRuntime.waitUntil` to run a background task and using `beforeunload` event to be notified when the instance is about to be shut down.
```

**Review by charislam — APPROVED** [review]

**Pull request merged**